### PR TITLE
Add a general sensor postprocessor, #25590 

### PR DESCRIPTION
**Pull request by harterj**
## Reason
This adds a basic generalized sensor postprocessor, and a thermocouple specific postprocessor. Users may desire to add sensors to multiphysics simulations to probe quantities such as fission power, temperature, pressure, or others. 

## Design
These postprocessors take signal input from another postprocessor, applies a Gaussian distribution to account for noise, also includes sensor delay and drift functions. They are designed to be upgraded in the future when a postdoc from an LDRD takes over.

## Impact
This adds a new feature.

Addresses issue #25590, tag @snschune @sallustius

### CODE REVIEW

File: framework/doc/content/source/postprocessors/ThermocoupleSensorPostprocessor.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# ThermocoupleSensorPostprocessor
+
+## Syntax and Description
+
+!syntax description /Postprocessors/ThermocoupleSensorPostprocessor
+
+`ThermocoupleSensorPostprocessor` models very specific classes of thermocouples,
```
Comment (GiudGiud) [inline framework/doc/content/source/postprocessors/ThermocoupleSensorPostprocessor.md:7]: I think this belongs in the misc module. 

unless you are using it in a particular application? if so which one?

File: framework/include/postprocessors/GeneralSensorPostprocessor.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "GeneralPostprocessor.h"
+
+/**
+ * A generalized sensor Postprocessor
```
Comment (GiudGiud) [inline framework/include/postprocessors/GeneralSensorPostprocessor.h:15]: need to be more descriptive than the name

File: framework/include/postprocessors/ThermocoupleSensorPostprocessor.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "GeneralSensorPostprocessor.h"
+
+/**
+ * A thermocouple Postprocessor
```
Comment (GiudGiud) [inline framework/include/postprocessors/ThermocoupleSensorPostprocessor.h:15]: same comment

File: framework/src/postprocessors/GeneralSensorPostprocessor.C [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+  // Sample a normal distribution centered around the noise mean, with a width of the noise standard
+  // deviation
+  Real normal_value = 0.0;
+  normal_value = MooseRandom::randNormal(_noise_mean, _noise_std_dev);
+
+  // If the timestep value is less than or equal to the sensor delay, use the previous value of the
+  // postprocessor
+  if (_dt <= _delay)
+    return _pp_old;
```
Comment (GiudGiud) [inline framework/src/postprocessors/GeneralSensorPostprocessor.C:60]: this doesnt seem right
if dt < delay all the time, the sensor will always be exact

File: framework/src/postprocessors/ThermocoupleSensorPostprocessor.C [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+registerMooseObject("MooseApp", ThermocoupleSensorPostprocessor);
+
+InputParameters
+ThermocoupleSensorPostprocessor::validParams()
+{
+  InputParameters params = GeneralSensorPostprocessor::validParams();
+  MooseEnum thermocoupleType("B E J K", "B");
+  params.addParam<MooseEnum>("thermocouple_type", thermocoupleType, "Thermocouple type");
+  params.addClassDescription("This is a ThermocoupleSensorPostprocessor for various classes of "
+                             "thermocouples, described by the 'thermocouple_type' parameter");
```
Comment (GiudGiud) [inline framework/src/postprocessors/ThermocoupleSensorPostprocessor.C:21]:
```suggestion
  params.addClassDescription("Models various classes of thermocouples defined by a NIST standard");
```

File: framework/src/postprocessors/ThermocoupleSensorPostprocessor.C [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  {
+    // Switch cases based on thermocouple temperature ranges
+    case ThermocoupleType::B:
+    {
+      if (pp_value < 273.15 || pp_value > 2093.15)
+        mooseWarning("Temperature is outside this thermocouple type's operation range.");
+    }
+    case ThermocoupleType::E:
+    {
+      if (pp_value < -543.15 || pp_value > 1273.15)
```
Comment (GiudGiud) [inline framework/src/postprocessors/ThermocoupleSensorPostprocessor.C:55]: this hard codes a unit
and
-543.15 is below the absolute 0? what's going on, which unit is this?

**Review by GiudGiud — CHANGES_REQUESTED** [review]

**Comment by snschune** [comment]
Summary from a discussion last week:
1. The noise should be randomly sampled from a Gaussian.
2. Delay will require to store history. 
3. We need to allow "continuous" as in the example of a thermocouple.
4. We should allow the user to specify a response function (preferable as a MOOSE function that is provided to the object).

**Pull request closed without merging**